### DISCUSSION TITLE
How to correctly use third-party GraphQL schema object types

### DISCUSSION THREAD

**Discussion by tony-oldport** [MichalLytek/type-graphql#1391] [Q&A] [answered]
I'm using a library [graphql-geojson](https://github.com/ghengeveld/graphql-geojson) that defines a number of [GraphQL schema object types](https://github.com/ghengeveld/graphql-geojson/blob/master/index.js) based on the GeoJSON spec.

I thought (probably naively) that I may just be able to import these and use them as a Field type, but I get the `CannotDetermineGraphQLTypeError: Cannot determine GraphQL output type` error.

The error only seems to happen when I use one of the `GraphQLObjectType` like `MultiPolygonObject`. No problems when I try one of the `GraphQLScalarType` like `CoordinatesScalar`

I've also tried adding the types to `orphanedTypes` with no luck.

```
import { Field, ObjectType } from 'type-graphql'
import { MultiPolygonObject } from 'graphql-geojson'

@ObjectType()
export class Zone {
  @Field(type => MultiPolygonObject, { nullable: true })
    multiPolygon?: any
}
```

**Comment by MichalLytek** [comment, marked as answer]
AFAIR TypeGraphQL does not cooperate with `graphql-js` types. You would need to redeclare those types with decorators, maybe this tool can help you:
https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-type-graphql

The support could be easily added, but I don't have throughput for that for now 😕

**Reply by tony-oldport** [reply]
Ahh, gotcha. Appreciate it as always.
